docs/Ignoring_files_from_build.md

```diff
@@ -0,0 +1,49 @@
+# Ignoring files from mbed build
+
+This is a documentation for `.mbedignore` file. It allows you to ignore files and directories from being processed by `mbed build` command.
+
+## Usage
+You can place the `.mbedignore` file in any directory where `mbed build` command is going to search for source files.
+
+Most convenient place is the root directory of the library or application, however this is not a requirement.
+
+Avoid defining rules that would cross the library boundaries as those would lead to side effects or build problems that are hard to find.
+
+## Syntax
+
+Each line in `.mbedignore` file is a file pattern used for matching files. Each matched file or directory is ignored from build.
+
+Following wildcards are accepted:
+
+|Pattern | Meaning|
+|--------|--------|
+| `*` | matches everything |
+| `?` | matches any single character |
+| `[seq]` | matches any character in seq |
+| `[!seq]` |matches any character not in seq|
+
+File is parsed with Python's [fnmatch](https://docs.python.org/2/library/fnmatch.html) functionality so syntax follows basic shell patterns with following exceptions:
+
+1. Each line is internally prefixed with the path of the `.mbedignore` file.
+2. Line cannot start with `.` or `/` (because of rule 1)
+
+Globbing functionality is not used, so you cannot recursively match specific file pattern. You need to define rule per directory in that case.
```

Comment:
The globbing functionality is not used, so you cannot recursively match a specific file pattern. In such case, you need to define a rule for each directory.